test/router/router2.test.lua

```diff
@@ -46,6 +46,7 @@ vshard.router.cfg(cfg)
 f7 = vshard.router.static.discovery_fiber
 vshard.router.static.discovery_fiber:status()
 
+while f6:status() ~= 'dead' do fiber.sleep(0.01) end
 f1:status(), f2, f3:status(), f4:status(), f5, f6:status(), f7:status()
```

Comment:
That should help to fix the exact problem of the ticket, but I also see another potential issue: by the time we see that `f6` is dead, `f7` can also become dead. This is because it works with `discovery_mode = 'once'`. The fiber will cancel itself when all is discovered.
I suggest to fix it too. For example, like this:
```Lua
assert(f7:status() == 'suspended' or vshard.router.info().bucket.unknown == 0)
```

Reply:
Yep, that may be a problem. Fixed